renderer/containers/cropper.js

```diff
@@ -330,7 +356,56 @@ export default class CropperContainer extends Container {
       this.setBounds({x, y, width, height});
       this.setState({isMoving: false, showHandles: true});
       this.cursorContainer.removeCursorObserver(this.move);
-      this.updateSettings({x, y});
+      // Check: this.updateSettings({x, y}); <-- do we need this? since it alreadly calls setBounds which updates Settings
+    }
+  };
+
+  redo = () => {
+    if (this.state.isUndone && this.state.cropperPointer < this.state.cropperHistory.length - 1) {
+      const originalCropperHistory = this.state.cropperHistory;
+      const currentPointer = this.state.cropperPointer + 1;
+      const newState = this.state.cropperHistory[currentPointer];
+      this.setState(newState);
+      this.setState({
+        cropperHistory: originalCropperHistory,
+        cropperPointer: currentPointer,
+        isUndone: true
+      });
+
+      const {x, y, width, height, ratio, displayId} = this.state;
+
+      this.settings.set('cropper', {
+        x,
+        y,
+        width,
+        height,
+        ratio,
+        displayId
+      });
+    }
+  };
+
+  undo = () => {
```

Comment:
There's a lot of duplicated logic between the `undo` and `redo` methods. I would try to unify the logic into an internal method which both of these call.